File: app/src/main/java/com/android/ichooseyou/util/Alert.java
package com.android.ichooseyou.util;

import android.content.Context;
import android.content.DialogInterface;

import androidx.appcompat.app.AlertDialog;

public class Alert {
    public enum AlertType {
        INFORMATION, WARNING, ERROR, CONFIRMATION;
    }

    private final AlertDialog.Builder builder;

    public Alert(Context context, Alert.AlertType alertType) {
        builder = new AlertDialog.Builder(context);

        switch (alertType) {
            case INFORMATION:
                builder.setTitle("Information");
                break;

            case WARNING:
                builder.setTitle("Warning");
                break;

            case ERROR:
                builder.setTitle("Error");
                break;

            case CONFIRMATION:
                builder.setTitle("Confirmation");
                break;
        }
    }

    public Alert setContent(String message) {
        builder.setMessage(message);
        return this;
    }

    public Alert setPositiveButton(String text, DialogInterface.OnClickListener listener) {
        builder.setPositiveButton(text, listener);
        return this;
    }

    public Alert setNegativeButton(String text, DialogInterface.OnClickListener listener) {
        builder.setNegativeButton(text, listener);
        return this;
    }

    public Alert setNeutralButton(String text, DialogInterface.OnClickListener listener) {
        builder.setNeutralButton(text, listener);
        return this;
    }

    public void show() {
        builder.create().show();
    }
}

File: app/src/main/java/com/android/ichooseyou/util/Alert.kt
package com.android.ichooseyou.util

import android.content.Context
import android.content.DialogInterface
import androidx.appcompat.app.AlertDialog

class Alert(context: Context?, alertType: AlertType?) {
    enum class AlertType {
        INFORMATION,
        WARNING,
        ERROR,
        CONFIRMATION
    }

    private val builder: AlertDialog.Builder

    init {
        builder = AlertDialog.Builder(context!!)
        when (alertType) {
            AlertType.INFORMATION -> builder.setTitle("Information")
            AlertType.WARNING -> builder.setTitle("Warning")
            AlertType.ERROR -> builder.setTitle("Error")
            AlertType.CONFIRMATION -> builder.setTitle("Confirmation")
            null -> TODO()
        }
    }

    fun setContent(message: String?): Alert {
        builder.setMessage(message)
        return this
    }

    fun setPositiveButton(text: String?, listener: DialogInterface.OnClickListener?): Alert {
        builder.setPositiveButton(text, listener)
        return this
    }

    fun setNegativeButton(text: String?, listener: DialogInterface.OnClickListener?): Alert {
        builder.setNegativeButton(text, listener)
        return this
    }

    fun setNeutralButton(text: String?, listener: DialogInterface.OnClickListener?): Alert {
        builder.setNeutralButton(text, listener)
        return this
    }

    fun show() {
        builder.create().show()
    }
}
